src/main/scala/io/iohk/ethereum/ledger/InMemoryWorldStateProxy.scala

```diff
@@ -155,7 +155,8 @@ class InMemoryWorldStateProxy private(
     copyWith(accountsStateTrie = accountsStateTrie.put(address.bytes, account))
 
   override def deleteAccount(address: Address): InMemoryWorldStateProxy =
-    copyWith(accountsStateTrie = accountsStateTrie.remove(address.bytes))
+    copyWith(accountsStateTrie = accountsStateTrie.remove(address.bytes),
+      contractStorages = contractStorages - address)
```

Comment:
I evaluated deleting the code from `evmCodeStorage` of contracts when deleting accounts in `Ledger.deleteAccouts` but I don't think it would work properly if the code is duplicated among different contracts (for example as mentioned in https://www.reddit.com/r/ethereum/comments/571j84/lets_say_i_want_to_make_millions_of_contracts/). I left a comment regarding this in https://github.com/input-output-hk/etc-client/blob/phase/3/txExecution/src/main/scala/io/iohk/ethereum/ledger/Ledger.scala#L380